src/state-manager/TransactionConsensus.ts

```diff
@@ -3575,183 +3574,207 @@ class TransactionConsenus {
+      //     // Gossip the vote to the entire consensus group
+      //     // Comms.sendGossip('gossip-applied-vote', ourVote, '', null, filteredConsensusGroup, true, 4, queueEntry.acceptedTx.txId, `${NodeList.activeIdToPartition.get(ourVote.node_id)}`)
+      //   } else {
+      //     this.profiler.profileSectionStart('createAndShareVote-tell')
+      //     // if (this.config.p2p.useBinarySerializedEndpoints && this.config.p2p.spreadAppliedVoteHashBinary) {
+      //       const request = appliedVoteHash as AppliedVoteHash
+      //       this.p2p.tellBinary<SpreadAppliedVoteHashReq>(
+      //         filteredConsensusGroup,
+      //         InternalRouteEnum.binary_spread_appliedVoteHash,
+      //         request,
+      //         serializeSpreadAppliedVoteHashReq,
+      //         {}
+      //       )
+      //     // } else {
+      //       // this.p2p.tell(filteredConsensusGroup, 'spread_appliedVoteHash', appliedVoteHash)
+      //     // }
+
+      //     this.profiler.profileSectionEnd('createAndShareVote-tell')
+      //   }
+      // } else {
+      //   nestedCountersInstance.countEvent('transactionQueue', 'createAndShareVote fail, no consensus group')
+      // }
     } catch (e) {
       this.mainLogger.error(`createAndShareVote: error ${e.message}`)
     } finally {
       this.profiler.profileSectionEnd('createAndShareVote', true)
     }
   }
 
-  calculateVoteHash(vote: AppliedVote, removeSign = true): string {
-    if (this.stateManager.transactionQueue.usePOQo) {
+  calculateVoteHash(vote: AppliedVote | Proposal, removeSign = true): string {
+    if (this.stateManager.transactionQueue.usePOQo && (vote as Proposal).applied !== undefined) {
+      const proposal = vote as Proposal
+      const applyStatus = {
+        applied: proposal.applied,
+        cantApply: proposal.cant_preApply,
+      }
+      const accountsHash = this.crypto.hash(
+        this.crypto.hash(proposal.accountIDs) +
+        this.crypto.hash(proposal.beforeStateHashes) + 
+        this.crypto.hash(proposal.afterStateHashes)
+      )
+      const proposalHash = this.crypto.hash(
+        this.crypto.hash(applyStatus) + accountsHash + proposal.appReceiptDataHash
+      )
+      return proposalHash
+    } else if (this.stateManager.transactionQueue.usePOQo) {
+      const appliedVote = vote as AppliedVote
       const appliedHash = {
-        applied: vote.transaction_result,
-        cantApply: vote.cant_apply
+        applied: appliedVote.transaction_result,
+        cantApply: appliedVote.cant_apply
       }
       const stateHash = {
-        account_id: vote.account_id,
-        account_state_hash_after: vote.account_state_hash_after,
-        account_state_hash_before: vote.account_state_hash_before,
+        account_id: appliedVote.account_id,
+        account_state_hash_after: appliedVote.account_state_hash_after,
+        account_state_hash_before: appliedVote.account_state_hash_before,
```

Comment:
Oh actually we never reach this else if statement. In the current POQ setup with the new types (Proposal), we will only ever need the first if statement 

`if (this.stateManager.transactionQueue.usePOQo && (vote as Proposal).applied !== undefined)`

But actually, I'll comment out the other else if and else statements since they're pretty much deprecated now.